src/main/java/edu/utdallas/davisbase/storage/TableFile.java

```diff
@@ -185,22 +190,6 @@ public void appendRow(TableRowBuilder tableRowBuilder) throws IOException {
 
 		short dataEntryPoint = (short) (cellOffset - totalSpaceRequired);
```

Comment:
This type of cast is dangerous. If there is an overflow/underflow in the cast, then it will go undetected and our system will be in an invalid state for an unknown reason. Better to "fail fast".

```suggestion
		short dataEntryPoint = com.google.common.primitives.Shorts.checkedCast(cellOffset - totalSpaceRequired);
```